packages/@lwc/template-compiler/src/shared/ast.ts

```diff
@@ -316,6 +335,15 @@ export function dynamicDirective(value: Expression, location: SourceLocation): D
     };
 }
 
+export function isDirective(value: Expression, location: SourceLocation): IsDirective {
```

Comment:
Might be worth adding a comment here. There's a lot of `isFoo` in `@lwc/template-compiler`, which usually has return type `node is Foo`. Example: `isBaseElement` returns type `node is BaseElement`.

Helpful that you include an explicit return type here, but a comment would help to prevent any remaining confusion.

Alternately, you could rename to `lwcIsDirective` to disambiguate.